src/main/java/appeng/helpers/PatternHelper.java

```diff
@@ -85,7 +85,13 @@ public PatternHelper( final ItemStack is, final World w )
 
 		for( int x = 0; x < inTag.tagCount(); x++ )
 		{
-			final ItemStack gs = new ItemStack( inTag.getCompoundTagAt( x ) );
+			NBTTagCompound ingredient = inTag.getCompoundTagAt( x );
+			final ItemStack gs = new ItemStack( ingredient );
+
+			if( !ingredient.hasNoTags() && gs.isEmpty() )
+			{
+				throw new IllegalArgumentException( "No pattern here!" );
```

Comment:
Okay, maybe rephrasing helps conveying, why I think, it is a valid change.

In the [loop](https://github.com/tyra314/Applied-Energistics-2/blob/f303f9fe8c5c82738b0e3a80cd1fab65f5ce4f47/src/main/java/appeng/helpers/PatternHelper.java#L86), `inTag` is always 9 fields, each representing on slot of a crafting grid.

For instance: `inTag` could be like this `[{},{},{},{},{id:"minecraft:log",Count:1b,Damage:0s},{},{},{id:"inca:llama_spit_bottle",Count:1b,Damage:0s},{}]`

Therefore, it is totally fine for a valid EncodedPattern to contain empty ItemStacks. However, trying to construct an ItemStack with an invalid NBTCompound also results in an empty ItemStack. Also, I'm not aware of another way to test an NBTCompound, if it's an valid ItemStack, other than just creating the ItemStack with it. So I have resorted to create the ItemStack and checking, whether it's empty, but the NBTCompound wasn't. If this happens, there is now way, you could instantiate the same encodedPattern like it was stored before. In that case, I throw the exception to get into the invalid encodedPattern code path, just like in other cases.